mlflow/pipelines/steps/train.py

```diff
@@ -652,3 +725,27 @@ def _construct_search_space_from_yaml(self, params):
                     error_code=INVALID_PARAMETER_VALUE,
                 )
         return search_space
+
+    def _write_yaml_output(self, best_hp_params, best_hardcoded_params, output_directory):
+        best_parameters_path = os.path.join(output_directory, "best_parameters.yaml")
+        if os.path.exists(best_parameters_path):
+            os.remove(best_parameters_path)
+        with open(best_parameters_path, "a") as file:
+            file.write("# tuned hyperparameters\n")
+            self._process_and_safe_dump(best_hp_params, file, default_flow_style=False)
+            file.write("# hardcoded parameters\n")
+            self._process_and_safe_dump(best_hardcoded_params, file, default_flow_style=False)
+        mlflow.log_artifact(best_parameters_path, artifact_path="best_parameters.yaml")
+
+    def _process_and_safe_dump(self, data, file, **kwargs):
+        import numpy as np
+
+        processed_data = {}
+        for key, value in data.items():
+            if isinstance(value, np.floating):
```

Comment:
@dbczumar `np.floating` covers various precisions of numpy floats 

https://stackoverflow.com/questions/28292542/how-to-check-if-a-number-is-a-np-float64-or-np-float32-or-np-float16

Reply:
wonderful! let's make sure to include a test case for this kind of behavior in our tests PR